src/fmeval/eval_algorithms/answer_relevance.py

```diff
@@ -0,0 +1,289 @@
+
+
+@dataclass(frozen=True)
+class AnswerRelevanceConfig(EvalAlgorithmConfig):
+    """Configures the Answer Relevance evaluation algorithm.
+
+    :param strictness: The number questions generated per answer.
+    """
+
+    strictness: Optional[int] = 1
+
+
+class AnswerRelevanceScore(Transform):
+    """This transform augments its input record with the computed answer relevance score.
+
+    See the docstring for `Answer Relevance` for more details regarding the score itself.
+    """
+
+    def __init__(
+        self,
+        embeddings_model: ModelRunner,
+        model_output_key: str = DatasetColumns.MODEL_OUTPUT.value.name,
+        output_key: str = ANSWER_RELEVANCE,
+    ):
+        """AnswerRelevanceScore initializer.
+
+        :param embeddings_model: An instance of ModelRunner representing the embedding model to be used.
+        :param model_output_key: The key corresponding to the model output.
+        :param output_key: The key corresponding to the answer relevance score that will be added to the input record.
+        """
+        super().__init__(embeddings_model, model_output_key, output_key)
+        self.register_input_output_keys(
+            input_keys=[GEN_QUESTIONS, model_output_key],
+            output_keys=[output_key],
+        )
+        self.output_key = output_key
+        self.model_output_key = model_output_key
+        self.embeddings_model = embeddings_model
+
+    @validate_call
+    def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
+        """Augment the input record with the computed answer relevance score.
+
+        :param record: The input record.
+        :returns: The input record, with the answer relevance score added in.
+        """
+        gen_questions_str = record[GEN_QUESTIONS]
+        question = record[self.model_output_key]
+        record[self.output_key] = self._get_score(question, gen_questions_str)
+        return record
+
+    def _get_score(self, question: str, gen_questions_str: str) -> float:
+        """Given generated questions and the original question, embedding model, computer Answer Relevance score.
+
+        :param question: The given prompt (model input).
+        :param gen_questions_str: Questions generated by judge model based on the answer.
+        :returns: 0 to 1. See the docstring for `Answer Relevance` for more details
+            on what these numerical values represent.
+        """
+        question_vector = np.asarray(self.embeddings_model.predict(question))
+        question_norm = np.linalg.norm(question_vector)
+        total = 0
+        if gen_questions_str:
+            generated_questions = gen_questions_str.split("\n")
+            for generated_question in generated_questions:
+                gen_question_vector = np.asarray(self.embeddings_model.predict(generated_question))
+                norm = np.linalg.norm(gen_question_vector) * question_norm
+                total += np.dot(gen_question_vector, question_vector.T) / norm
```

Comment:
Sorry I didn't catch this earlier, but the second vector shouldn't be transposed. Both embeddings should be of the same dimension, in which case the standard code for computing the dot product is just `np.dot(x, y)` where both vectors are row vectors. Numpy has a rather unintuitive behavior where it will actually give you the correct dot product even when one of the vectors is transposed, which is why the unit tests are passing.

But since this behavior is counterintuitive, having the transpose is confusing for those reading the code, as it looks like you're computing an outer product instead (row vector multiplied by column vector).